app/src/main/java/com/firebase/ecommerce/core/StoreData.kt

```diff
@@ -23,4 +23,10 @@ class StoreData @Inject constructor( private val dataStore: DataStore<Preference
         }
     }
 
+    suspend fun clearUserData() {
+        dataStore.edit { preferences ->
+            preferences[USERNAME] = ""
```

Comment:
@Ankita123-oss Instead of editing Username preference to empty string, try to clear that particular field from datastore